tests/unit/modules/test_telegram.py

```diff
@@ -22,42 +24,73 @@
 import salt.modules.telegram as telegram
 
 
+log = logging.getLogger(__name__)
+
+
+class RequestMock(Mock):
+    '''
+    Request Mock
+    '''
+
+    def get(self, *args, **kwargs):
+        return RequestResponseMock()
+
+    def put(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+        return RequestPutResponseMock()
+
+    def delete(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+        return RequestResponseMock()
+
+
+class RequestResponseMock(Mock):
+    '''
+    Request Response Mock
+    '''
+    def json(self):
+        return [{'url': 'http://example.org',
+                 '_id': 1234}, ]
+
+
+class RequestPutResponseMock(Mock):
+    '''
+    Request Put Response Mock
+    '''
+    ok = True
+
+    def json(self):
+        return {'_id': 4321}
+
+
+REQUEST_MOCK = RequestMock()
```

Comment:
Lets try and avoid creating unnecessary globals in test modules.
If the reason is to only create this once, then lets do it in `setUpClass`